sanic/mixins/startup.py

```diff
@@ -892,7 +892,8 @@ def get_address(
             Tuple[str, int]: Tuple containing the host and port
         """  # noqa: E501
         host = host or "127.0.0.1"
-        port = port or (8443 if (version == 3 or auto_tls) else 8000)
+        if port is None:
```

Comment:
I am generally fine with this approach, but I think maybe we should be explicit.

```python
if port is None:
    # use a default
elif port is 0:
    port = _get_random_port()
...
```

In this case we would bind to get something from the os, but immediately close. I suppose in theory this might fail for some interval, would need to explore that. But I think we want to ensure that `get_address` does indeed return `tuple[str, int]`. If not I suspect the changes might need to be deeper here.